lib/catalog/submit_order_item.rb

```diff
@@ -11,21 +11,21 @@ def initialize(order_id)
     def process
       @order = Order.find_by!(:id => @order_id)
 
-      # TODO: This will need to be updated to figure out the next order item in the
-      # process sequence to run. For now, we will only submit the 'applicable' order
-      # item.
-      @order.order_items.where(:process_scope => 'applicable').each do |order_item|
-        raise ::Catalog::NotAuthorized unless valid_source?(order_item.portfolio_item.service_offering_source_ref)
+      # For now simply submit next order_item after previous one is completed
+      order_item = @order.order_items.find(&:can_order?)
+      return self unless order_item
 
-        if ::Catalog::SurveyCompare.any_changed?(order_item.portfolio_item.service_plans)
-          order_item.mark_failed("Order Item Failed: Base survey does not match Topology")
-          raise ::Catalog::InvalidSurvey, "Base survey does not match Topology"
-        end
+      raise ::Catalog::NotAuthorized unless valid_source?(order_item.portfolio_item.service_offering_source_ref)
 
-        submit_order_item(order_item)
-
-        Rails.logger.info("Order #{@order_id} submitted for provisioning")
+      if ::Catalog::SurveyCompare.any_changed?(order_item.portfolio_item.service_plans)
+        order_item.mark_failed("Order Item Failed: Base survey does not match Topology")
+        raise ::Catalog::InvalidSurvey, "Base survey does not match Topology"
       end
+
+      submit_order_item(order_item)
+
+      Rails.logger.info("Order Item #{order_item.id} submitted for provisioning")
+
       @order.update(:state => 'Ordered', :order_request_sent_at => Time.now.utc)
```

Comment:
I forgot to mention this but `order_request_sent_at` is going to be "wrong" now, since the initial "before" will update this to be when the first order_item request is sent, but then the "applicable" will update this again, and the "after" will update it as well.

I'm not sure the best solution here, but it feels like we will lose a bit of clarity if we keep it how it is now.

Reply:
Update only if state is not `Ordered`